### ISSUE TITLE
[Request] Automatic operations

### ISSUE THREAD

**Issue by DaZombieKiller** [sailro/Reflexil#70]
I'm not all that great with introductions, so I'll jump right into it. 
Imagine I have a .NET dll, and I want to apply the same operations to it every time a new version comes out, but it takes a significant amount of time to do so.

Would it be possible for Reflexil to introduce a scripting language of sorts, for automatically executing a series of operations like adding references, renaming/injecting methods and fields, etc?

**Comment by Splamy** [comment]
Hey, maybe I could interest you in my (somewhat abandoned) ilpatcher I wrote once but never finished: https://github.com/Splamy/ILPatcher
It's probably not (yet) as convenient as reflexil but with some work it might get a reasonable patcher.

**Comment by sailro** [comment]
Yes I always wanted to have something to "replay" in a logical manner all alterations done through the GUI.

**Comment by DaZombieKiller** [comment]
Alternatively, the ability to use Reflexil as a general-purpose C# library could work too, if that isn't supported already

**Comment by beatcracker** [comment]
@DaZombieKiller This is what I did, when I needed to [patch the assembly programmatically](https://github.com/beatcracker/Ori-Controller-Remap): https://github.com/beatcracker/GUILess-Reflexil

**Comment by DaZombieKiller** [comment]
I ended up just writing my own program using Mono.Cecil to get the job done, but if I ever need to do something much more complex I'll definitely take a look at that.

**Comment by beatcracker** [comment]
@DaZombieKiller It's outdated and I'm Ops guy, not Dev. But it's something I'd like to see in Reflexil: standalone NuGet package for easy IL hacking.
